JsonLogic/RuleRegistry.cs

```diff
@@ -10,12 +16,55 @@ namespace Json.Logic;
 /// </summary>
 public static class RuleRegistry
 {
-	private static readonly Dictionary<string, Type> _rules =
-		typeof(Rule).Assembly
-			.GetTypes()
-			.Where(t => typeof(Rule).IsAssignableFrom(t) && !t.IsAbstract)
-			.SelectMany(t => t.GetCustomAttributes<OperatorAttribute>().Select(a => new { Name = a.Name, Type = t }))
-			.ToDictionary(t => t.Name, t => t.Type);
+	private static readonly ConcurrentDictionary<string, Type> _rules;
+	private static readonly ConcurrentDictionary<Type, JsonSerializerContext> _externalRuleResolvers = new();
+
+#if NET8_0_OR_GREATER
+	internal static IJsonTypeInfoResolver[] ExternalTypeInfoResolvers => _externalRuleResolvers.Values.Distinct().ToArray();
+#endif
+
+	static RuleRegistry()
+	{
+		_rules = new ConcurrentDictionary<string, Type>(new Dictionary<string, Type>
```

Comment:
Didn't one of the rules have two operator aliases for it? Eyeballing this I don't see any double types in the list.

Reply:
Yeah, it's the if rule. It's there.